jupyter_theme_editor/handlers.py

```diff
@@ -0,0 +1,42 @@
+import json
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.utils import url_path_join
+import tornado
+from jinja2 import Template
+
+
+class RouteHandler(APIHandler):
+    # The following decorator should be present on all verb methods (head, get, post,
+    # patch, put, delete, options) to ensure only authorized user can request the
+    # Jupyter server
+
+    @tornado.web.authenticated
+    def post(self):
+        # input_data is a dictionary with a key "name"
+        input_data = self.get_json_body()
+        new_input_data = {}
+        for key, value in input_data.items():
+            value = str(input_data[key])
+            new_value = value.replace(' ', '')
+            new_key = key.replace('--jp-', '').replace('-', '_')
+            new_input_data[new_key] = new_value
+
+
+        with open('style/template.css', 'r') as file:
+            template = file.read()
+            self.log.info('template is:', template)
+
+        j2_template = Template(template)
```

Comment:
For testing, this is good enough. But it won't work well when distributing the extension as a Python package.

You should use a [loader](https://jinja.palletsprojects.com/en/3.0.x/api/#loaders). The easiest being something like in the [example](https://jinja.palletsprojects.com/en/3.0.x/api/#basics).

```python
from jinja2 import Environment, PackageLoader
env = Environment(
    loader=PackageLoader("jupyter_theme_editor", "templates")
)
j2_template = env.get_template("template.css")
```

> This implies that you move the file `style/template.css` in `jupyter_theme_editor/templates/template.css`.